src/lib/viewers/media/MediaBaseViewer.js

```diff
@@ -680,16 +724,64 @@ class MediaBaseViewer extends BaseViewer {
      * @return {void}
      */
     addEventListenersForMediaElement() {
-        this.mediaEl.addEventListener('timeupdate', this.setTimeCode);
-        this.mediaEl.addEventListener('progress', this.progressHandler);
-        this.mediaEl.addEventListener('volumechange', this.updateVolumeIcon);
-        this.mediaEl.addEventListener('playing', this.playingHandler);
+        this.mediaEl.addEventListener('ended', this.mediaendHandler);
         this.mediaEl.addEventListener('pause', this.pauseHandler);
-        this.mediaEl.addEventListener('ended', this.resetPlayIcon);
+        this.mediaEl.addEventListener('playing', this.playingHandler);
+        this.mediaEl.addEventListener('progress', this.progressHandler);
         this.mediaEl.addEventListener('seeked', this.seekHandler);
-        this.mediaEl.addEventListener('ended', this.mediaendHandler);
+        this.mediaEl.addEventListener('timeupdate', this.setTimeCode);
+        this.mediaEl.addEventListener('volumechange', this.updateVolumeIcon);
+    }
+
+    /**
+     * Removes event listeners to the media element
+     * @return {void}
+     */
+    removeEventListenersForMediaElement() {
+        this.mediaEl.removeEventListener('canplay', this.handleCanPlay);
+        this.mediaEl.removeEventListener('ended', this.mediaendHandler);
+        this.mediaEl.removeEventListener('loadstart', this.handleLoadStart);
+        this.mediaEl.removeEventListener('pause', this.pauseHandler);
+        this.mediaEl.removeEventListener('playing', this.playingHandler);
+        this.mediaEl.removeEventListener('progress', this.progressHandler);
+        this.mediaEl.removeEventListener('seeked', this.seekHandler);
+        this.mediaEl.removeEventListener('timeupdate', this.setTimeCode);
+        this.mediaEl.removeEventListener('volumechange', this.updateVolumeIcon);
+    }
+
+    /**
+     * Callback from the 'loadstart' event from the media element. Triggers a timer to measure the initial buffer fill.
+     * @return {void}
+     */
+    handleLoadStart() {
+        const tag = Timer.createTag(this.options.file.id, MEDIA_METRIC.bufferFill);
```

Comment:
Is it worth creating a super basic helper method for this, like:

```
createTag(tagName) {
    return Timer.createTag(this.options.file.id, tagName);
}

Reply:
👍